src/main/kotlin/pt/up/fe/ni/website/backend/annotations/validation/ValidImage.kt

```diff
@@ -0,0 +1,47 @@
+package pt.up.fe.ni.website.backend.annotations.validation
+
+import jakarta.validation.Constraint
+import jakarta.validation.ConstraintValidator
+import jakarta.validation.ConstraintValidatorContext
+import jakarta.validation.Payload
+import org.springframework.web.multipart.MultipartFile
+import pt.up.fe.ni.website.backend.util.filenameExtension
+import kotlin.reflect.KClass
+
+@MustBeDocumented
+@Constraint(validatedBy = [ValidImageValidator::class])
+@Target(AnnotationTarget.VALUE_PARAMETER)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class ValidImage(
+    val message: String = "{files.invalid_image}",
+    val groups: Array<KClass<*>> = [],
+    val payload: Array<KClass<Payload>> = [],
+)
+
+class ValidImageValidator : ConstraintValidator<ValidImage, MultipartFile?> {
+
+    override fun isValid(value: MultipartFile?, context: ConstraintValidatorContext?): Boolean {
+
+        if (value == null) {
+            return true
+        }
+
+        if (!isSupportedContentType(value.contentType)) {
+            return false
+        }
+
+        if (!isSupportedFileExtension(value.filenameExtension())) {
+            return false
+        }
+
+        return true
+    }
+
+    private fun isSupportedContentType(contentType: String?): Boolean {
+        return contentType == "image/png" || contentType == "image/jpg" || contentType == "image/jpeg"
+    }
+
+    private fun isSupportedFileExtension(extension: String?): Boolean {
+        return extension == "png" || extension == "jpg" || extension == "jpeg"
+    }
```

Comment:
I think this is fine for now but in the future we could investigate a way to to support more image types and maybe a way to automatically convert and optimize them to a format like webp

Reply:
I think cloudinary supports these kinds of trasformations